cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/ProgramControllerServiceAdapter.java

```diff
@@ -80,13 +71,7 @@ public void running() {
       @Override
       public void failed(Service.State from, Throwable failure) {
         LOG.error("Program terminated with exception", failure);
-        if (propagateServiceError()) {
-          error(failure);
-        } else if (getState() != State.STOPPING) {
-          complete();
-        } else {
-          stop();
-        }
+        error(failure);
```

Comment:
If you expand below these comments (using the icon in the left-sdie, where the line numbers are), you can display the code below where you made your changes. On line 94/79 (old/new line numbers), it says:
```
if (getState() != State.STOPPING) {
```
Should that not be ``Service.State.STOPPING``?